internal/cutil/cslice.go

```diff
@@ -0,0 +1,92 @@
+package cutil
+
+// #include <stdlib.h>
+import "C"
+import "unsafe"
+
+// The following code needs some explanation:
+// This creates slices on top of the C memory buffers allocated before in
+// order to safely and comfortably use them as arrays. First the void pointer
+// is cast to a pointer to an array of the type that will be stored in the
+// array. Because the size of an array is a constant, but the real array size
+// is dynamic, we just use the biggest possible index value maxIdx, to make
+// sure it's always big enough. (Nothing is allocated by casting, so the size
+// can be arbitrarily big.) So, if the array should store items of myType, the
+// cast would be (*[maxIdx]myItem)(myCMemPtr).
+// From that array pointer a slice is created with the [start:end:capacity]
+// syntax. The capacity must be set explicitly here, because by default it
+// would be set to the size of the original array, which is maxIdx, which
+// doesn't reflect reality in this case. This results in definitions like:
+// cSlice := (*[maxIdx]myItem)(myCMemPtr)[:numOfItems:numOfItems]
+
+const maxIdx = 1<<31 - 1 // 2GB, max int32 value, should be safe
+const ptrSize = C.size_t(unsafe.Sizeof(unsafe.Pointer(nil)))
```

Comment:
We have a another approach to this in PR #408 (which I think you came up with) that's a lot shorter/easier to read IMO.

Reply:
This is outdated. ;-)